File: src/main/java/com/api/aumatch/domain/model/Ong.java
package com.api.aumatch.domain.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
public class Ong {


    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator = "increment")
    @Column(name="id_ong")
    private Long idOng;

    @Column(name="razao_social")
    private String razaoSocial;

    @Column
    private String cnpj;

    @OneToOne
    @JoinColumn(name = "id_endereco", nullable = false)
    private Endereco endereco;
    //private List<Chat> historicoChat;
    //private StatusOngEnum statusOngEnum;

}

File: src/main/kotlin/com/julianamaijer/aumatch/domain/model/Ong.kt
package com.julianamaijer.aumatch.domain.model

import com.julianamaijer.aumatch.enums.OngStatusEnum
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity(name="ong")
data class Ong (

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Int,

        @Column
        var companyName: String,

        @Column
        var registeredNumber: String,


        var animals: List<Animal>,

        var chats: List<Chat>,

        var address: Address,

        @Column
        var status: OngStatusEnum

        )
